### PR TITLE
Migrate CI to github actions

### LINKED ISSUE #680: CI refactor
Since prometheus-operator is now using github actions and we need PRs like https://github.com/prometheus-operator/kube-prometheus/pull/669 to test multiple k8s versions, I propose to follow prometheus-operator repo and move to github actions also in here.

Pros:
- [`matrix strategy`](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstrategy) allows for easy updates to test supported k8s versions without changing bash script logic.
- we can easily test our jsonnet generate pipeline on macos, same as in https://github.com/prometheus-operator/prometheus-operator/blob/master/.github/workflows/ci.yaml#L10-L23
- it is much easier to see which stage failed just by looking at PR and without going to external site
- all pipelines are separate allowing for having tests which aren't mandatory (for example we can test on previous k8s version, but not require it for merges)
- bash script hack are not necessary. Like ones from https://github.com/prometheus-operator/kube-prometheus/blob/master/tests/e2e/travis-e2e.sh or https://github.com/prometheus-operator/kube-prometheus/tree/master/scripts

Cons:
- someone needs to write it :smile: 

/cc @prometheus-operator/kube-prometheus-reviewers

### PR DESCRIPTION
**Pull request by dgrisonnet**
A running example can be found here: https://github.com/dgrisonnet/kube-prometheus/actions/runs/261350358

Fix #680 

cc @paulfantom

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    name: Unit tests
+    steps:
+    - uses: actions/checkout@v2
+    - run: make --always-make test
+  e2e-tests:
+    name: E2E tests
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        kind-image:
```
Comment (paulfantom) [inline .github/workflows/ci.yaml:34]: Can we somehow have this in `env` section? This would allow more visibility.

**Comment by paulfantom** [comment]
Apart from one nit it looks good to me :+1:

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (dgrisonnet) [reply .github/workflows/ci.yaml:34]: I could not get this to work, the env section seems to only allow string key-value pairs. Do you of a hack that could make this possible?

Comment (paulfantom) [reply .github/workflows/ci.yaml:34]: Nope, I don't have anything. Let's leave it as is if GH actions do not support it.

**Review by paulfantom — APPROVED** [review]
LGTM

**Pull request merged**